pkg/handler/repository_rpm.go

```diff
@@ -3,17 +3,58 @@ package handler
 import (
 	"net/http"
 
+	"github.com/content-services/content-sources-backend/pkg/api"
 	"github.com/content-services/content-sources-backend/pkg/dao"
 	"github.com/content-services/content-sources-backend/pkg/db"
 	"github.com/labstack/echo/v4"
 )
 
+const (
+	defaultSearchRpmLimit = 20
+)
+
+type RepositoryRpmHandler struct {
+	Dao dao.RpmDao
+}
+
 type RepositoryRpmRequest struct {
 	UUID string `param:"uuid"`
 }
 
-func RegisterRepositoryRpmRoutes(engine *echo.Group /*, rDao *dao.RepositoryRpmDao */) {
-	engine.GET("/repositories/:uuid/rpms", listRepositoryRpms)
+func RegisterRepositoryRpmRoutes(engine *echo.Group, rDao *dao.RpmDao) {
+	rh := RepositoryRpmHandler{
+		Dao: *rDao,
+	}
+	engine.GET("/repositories/:uuid/rpms", rh.listRepositoriesRpm)
+	engine.POST("/rpms/names", rh.searchRpmByName)
+}
+
+// searchRpmByName godoc
+// @Summary      Search RPMs
+// @ID           searchRpm
+// @Description  Search RPMs for a given list of repository URLs
+// @Tags         repositories,rpms
+// @Accept       json
+// @Produce      json
+// @Success      200 {object} api.SearchRpmRequest
+// @Router       /rpms/names [post]
+func (rh *RepositoryRpmHandler) searchRpmByName(c echo.Context) error {
+	_, orgId, err := getAccountIdOrgId(c)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
```

Comment:
shouldn't this be bad request instead of ISE?

Reply:
we've been using badIdentity for an error here: https://github.com/content-services/content-sources-backend/blob/main/pkg/handler/repositories.go#L80-L83